src/System.Linq/src/System/Linq/Take.cs

```diff
@@ -95,7 +95,7 @@ public static IEnumerable<TSource> TakeLast<TSource>(this IEnumerable<TSource> s
                 TakeLastIterator(source, count);
         }
 
-        private static IEnumerable<TSource> TakeLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        private static IEnumerable<TSource> TakeLastRegularIterator<TSource>(IEnumerable<TSource> source, int count)
```

Comment:
This will break the build when the .SizeOpt.cs is compiled in rather than the .SpeedOpt.cs

Reply:
Wasn't aware that UAP is configured for SizeOpt. Fixed this and improved Take.SizeOpt little.